### PR TITLE
Documentation: edits of Introduction

### PR DESCRIPTION
**Pull request by davidcok**
full edit of Introduction; minor changes to opening material and user guide

### CODE REVIEW

File: docs/_includes/Introduction.md [hunk 1]
```diff
@@ -1,12 +1,17 @@
...
 The Dafny static program verifier can be used to verify the functional
 correctness of programs.
+This document is a reference manual for the programming language and a user guide
+for the dafny tool that performns verification and compilation to an
+executable form.
 
 The Dafny programming language is designed to support the static
 verification of programs. It is imperative, sequential, supports generic
-classes, methods and functions, dynamic allocation, inductive and
+classes, inhewritance and abstraction, methods and functions, dynamic allocation, inductive and
```
Comment (RustanLeino) [inline docs/_includes/Introduction.md:14]:
```suggestion
classes, inheritance and abstraction, methods and functions, dynamic allocation, inductive and
```

File: docs/_includes/Introduction.md [hunk 2]
```diff
@@ -23,61 +28,69 @@ changing the program’s type declarations, specifications, and statements.
...
 by [Boogie](http://research.microsoft.com/boogie)
 [@Boogie:Architecture;@Leino:Boogie2-RefMan;@LeinoRuemmer:Boogie2]
 and [Z3](https://github.com/z3prover)[@deMouraBjorner:Z3:overview].
 
-From verified programs, the Dafny compiler produces code (`.dll` or
-`.exe`) for the .NET platform via intermediate C\# files. However, the
+From verified programs, the Dafny compiler can produce code for a number
+of different backends: javascript, go, java, or
+the .NET platform via intermediate C\# files. However, the
 facilities for interfacing with other .NET code are minimal.
```
Comment (RustanLeino) [inline docs/_includes/Introduction.md:47]: Let's list .NET first and capitalize the names of the others. Also, say there's partial support for C++ backend.

Replace the "However" sentence with something like "Each backend provides a basic Foreign Function Interface (called `:extern`). However, there is no automatic FFI generator, so `:extern` stubs need to be written by hand."

File: docs/_includes/Introduction.md (hunk 1 above)
Comment (davidcok) [reply docs/_includes/Introduction.md:14]: Fixed

File: docs/_includes/Introduction.md (hunk 2 above)
Comment (davidcok) [reply docs/_includes/Introduction.md]: Done

**Pull request merged**